### ISSUE TITLE
tab jumps are weird

### ISSUE THREAD

**Issue by otacke** [h5p/h5p-multi-choice#24]
It seems there might a problem with the tabindex. I have not really looked into this topic (related to ARIA), but it seems that the jump order for n alternatives is:

- 1st alternative
- 1s_ hint
- 2nd hint
- 3rd hint
- ...
- n-1th hint
- nth alternative
- nth hint

Looks strange to me.

**Comment by falcon-git** [comment]
You are right. With hints/tips it doesn't seem to work exactly as designed. Arrow keys to move the focus between alternatives is intended, tab to focus on the hint is also as designed. If you tab several times you move between hints, also as designed afaik. The last alternative gets the focus after the last hint, this is not as designed afaik.

**Comment by otacke** [comment]
Oh, now I get it. I still have to learn about how to design accessible applications.

I had assumed that the tab key was _the_ major key that should be used to jump from element to element which here are the alternatives. But it makes total sense to give focus to all the alternatives first and then choose the sub elements with the arrow keys. I guess since the role for the container says "radiogroup", people will know that they can now use the arrow keys for selection?

**Comment by falcon-git** [comment]
Yes, that is my understanding. The default behavior of normal radio button groups is that you use keys to select and tab will make you move along.

**Comment by icc** [comment]
Regarding the hints, I wonder if it would make more sense if you could **only** tab to the hint next to the focused option. It is essentially a list where you can possibly have an "infinite" number of tabbable buttons, which I believe is to be avoided according to ARIA.

**Comment by otacke** [comment]
Updating the issue:
Tab order is still odd.

1) Create a Multiple Choice question with 4 answer options, each one with a hint.
2) Don't select anyhing. Tab through content.
3) The tab order is:
   1) answer option `#1`
   2) hint of option `#1`
   3) hint of option `#2`
   4) hint of option `#3`
   5) answer option `#4`
   6) hint of option `#4`
4) Select the 1st answer option. Tab through content.
5) The tab order is:
   1) answer option `#1`
   2) hint of option `#1`
   3) hint of option `#2`
   4) hint of option `#3`
   5) hint of option `#4`
   Essentially the same as before, but now option `#4` is not tabbable
6) Select the 2nd answer option. Tab through content.
7) The tab order is:
   1) hint of option `#1`
   2) answer option `#2`
   3) hint of option `#2`
   4) hint of option `#3`
   5) hint of option `#4`

It kind of makes sense, in a way, once one of the options is selected, but it feels completely arbitrary before.

I'd go for @icc's suggestion from 2017 (2017!!! 😢) and only allow to tab to the hint if the answer option that it belongs to has focus.
